app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -9,17 +11,21 @@ data class Medicine(
     val quantity: Int = 0,
     val prescriptionStrength: String = "",
     val startDate: LocalDate = LocalDate.now(),
-    val prescriptionLength: String = "",
-    val time: Long = 0,
-    // Frequency per 24 hours
+    val prescriptionLength: Duration = Duration.ZERO,
+    val timeUnit: TimeUnit = TimeUnit.HOURS,
```

Comment:
- Changing the type of the prescriptionLength property to Duration instead of String will allow you to perform calculations on it easily and avoid any possible errors in the conversion process. You can use the Duration.parse() method to parse the String value to a Duration object.

- Using a TimeUnit enum instead of a Long for the time property will make the code more readable and avoid errors in the conversion process.

    Source: Source: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html

Reply:
Very nice.